### PR TITLE
fix: batch caching of remote pin lookups

### PR DESCRIPTION
**Pull request by rafaelramalho19**
Still WIP

- Refactored FilesList to functional components (speeds up the application loading & responsiveness, as well as allowing for this new optimization to work)
- Refactored all loadable components to use the new library (@loadable/component) so it doesn't throw errors in the console anymore due to latest version of react

### CODE REVIEW

**Review by lidel — CHANGES_REQUESTED** [review]
@rafaelramalho19 I've played a bit with this PR, and it mostly worked as expected: – no longer fetching all remote pins  :tada: :))

The only missing part was avoiding sending check for CIDs that were already probed, 
so I've added some optimizations in https://github.com/ipfs/ipfs-webui/pull/1741/commits/9792b476f68b9f648c53bbc8de5617ec64742787:
  - It  adds new cache for "not-pins" per each remote service, so we can remember remote pin state ("pinned" or "not pinned") of each item and avoid sending an unnecessary request every time we see the same CID anywhere in Files.

If you could take a look at remaining TODO here: 

- [ ] On the initial load, Files screen does not run the check for MFS root. One needs to go to a subdirectory to trigger remote pin checks, and then if one goes back to MFS root, the check is finally executed.  We are probably missing `doFetchRemotePins(<MFS root files>)` somehwere, or it gets skipped because some state is not ready yet.
- [ ] `ipld-explorer-components` still requires `react-loadable` so either you need to add it back for now, or refactor that lib as well.

**Comment by rafaelramalho19** [comment]
> On the initial load, Files screen does not run the check for MFS root. One needs to go to a subdirectory to trigger remote pin checks, and then if one goes back to MFS root, the check is finally executed. We are probably missing doFetchRemotePins(<MFS root files>) somehwere, or it gets skipped because some state is not ready yet.

Is this still happening? Can't replicate 🤔

**Comment by lidel** [comment]
I am able to reproduce this in both Firefox and Brave. 
There is no API call to `/api/v0/pin/remote/ls` the  very first time Files screen is opened. 
Do you have any remote pins in the root of Files?

**Review by lidel — APPROVED** [review]
Ok, managed to fix the problem. It manifested on my box because I use Firefox and have a loooot files in the root directory, so initial remote pin check was slow enough to trigger performance issues and race condition due to unnecessary rendering.

https://github.com/ipfs/ipfs-webui/pull/1741/commits/6207f68caf3ed27b0edc58b52d0ee29311287afd  simplifies the way we pass information about remote pins to cache by batching all pin updates into a single event. 
This way we avoid delays in UI caused by unnecessary re-renders.

Additional fixes:
- requests are only sent to services that are confirmed to be online
- CIDs to check are deduplicated (in case same item is present multiple times)
- remote pins are created with `background:true` 
  - we don't have UI for "ongoing pin" nor "failed pin" anyway
  - above can be added in the future, but it is  best-effort for now
- initial load of Files shows remote pins, but we fetch remote pins only if at least one service is online

@rafaelramalho19 looks ok to merge on my end, but  please take a look and give :+1:  if no concerns.

**Comment by lidel** [comment]
Ugh, I've merged  #1644   and now we need to resolve conflicts.
Merging master to this branch produced some weird errors around react, so rebase may be a safer way. 
Will look into it tomorrow.

**Comment by lidel** [comment]
@rafaelramalho19 resolved conflicts – give this a quick look and  :+1:  / :-1:  in a spare moment.

File: src/bundles/pinning.js [hunk 1]
```diff
@@ -140,33 +184,36 @@ const pinningBundle = {
...
-          dispatch({
-            type: 'ADD_REMOTE_PIN',
-            payload: {
-              ...pin,
-              id: `${service.name}:${pin.cid}`
-            }
-          })
-          await ipfs.pin.remote.add(cid, { service: service.name, name })
+          adds.push({ id, ...pin })
+          // TODO: remove background:true and add pin job to queue.
```
Comment (rafaelramalho19) [inline src/bundles/pinning.js:202]: Maybe let's add a github issue with this?

Comment (lidel) [reply src/bundles/pinning.js:202]: Yes, filled https://github.com/ipfs/ipfs-webui/issues/1751 and https://github.com/ipfs/ipfs-webui/issues/1752
(we should add error handling first, tracking progress will be more complex)

**Pull request merged**